src/docfx/template/TemplateEngine.cs

```diff
@@ -37,21 +35,19 @@ internal class TemplateEngine : IDisposable
             _config = config;
             _output = output;
 
-            _templateDir = config.Template.Type switch
+            var template = config.Template;
+            if (template.Type == PackageType.None)
             {
-                PackageType.None => Path.Combine(buildOptions.DocsetPath, "_themes"),
-                _ => packageResolver.ResolvePackage(config.Template, PackageFetchOptions.DepthOne),
-            };
-
-            _contentTemplateDir = Path.Combine(_templateDir, "ContentTemplate");
-            _schemaDir = Path.Combine(_contentTemplateDir, "schemas");
+                template = new PackagePath(Path.Combine(buildOptions.DocsetPath, "_themes"));
+            }
 
-            _templateDefinition = PathUtility.LoadYamlOrJson<TemplateDefinition>(errors, _templateDir, "template") ?? new TemplateDefinition();
+            _package = packageResolver.ResolvePackage2(template, PackageFetchOptions.DepthOne);
+            _templateDefinition = _package.TryReadYamlOrJson<TemplateDefinition>(errors, "template") ?? new TemplateDefinition();
```

Comment:
>_package.TryReadYamlOrJson<TemplateDefinition>(errors, "template") ?? new TemplateDefinition(); [](start = 34, length = 95)

make this a Lazy load since it is not always needed